molgenis-data-mysql/src/main/java/org/molgenis/data/mysql/MysqlRepository.java

```diff
@@ -678,7 +678,19 @@ public Entity findOne(Object id)
 	public Iterable<Entity> findAll(Iterable<Object> ids)
 	{
 		if (ids == null) return Collections.emptyList();
-		return findAll(new QueryImpl().in(getEntityMetaData().getIdAttribute().getName(), ids));
+		Query inQuery = new QueryImpl().in(getEntityMetaData().getIdAttribute().getName(), ids);
+		Map<Object, Entity> index = uniqueIndex(newArrayList(findAll(inQuery)), Entity::getIdValue);
+		return Iterables.filter(Iterables.transform(ids, id -> lookup(index, id)), x -> x != null);
+	}
+
+	public Entity lookup(Map<Object, Entity> index, Object id)
+	{
+		Entity result = index.get(id);
+		if (result == null)
+		{
+			LOG.warn("Couldn't find {} for id {}.", getName(), id);
```

Comment:
I ran into an issue here where the type of the ID that gets sent to the query method is different from the java type returned by the getIdValue of the entity. I think that it was a bug that ElasticSearch sent String representations of int IDs to the findAll method. But perhaps it happens elsewhere as well.